helix-view/src/handlers/lsp.rs

```diff
@@ -1 +1,40 @@
+use crate::{DocumentId, ViewId};
 
+#[derive(Debug, Clone, Copy)]
+pub struct CompletionTrigger {
+    /// The char position of the primary cursor when the
+    /// completion was triggered
+    pub trigger_pos: usize,
+    pub doc: DocumentId,
+    pub view: ViewId,
+    /// Whether the cause of the trigger was an automatic completion (any word
+    /// char for words longer than minimum word length).
+    /// This is false for trigger chars send by the LS
+    pub auto: bool,
+}
+
+pub enum CompletionEvent {
+    /// Auto completion was triggered by typing a word char
+    /// or a completion trigger
+    Trigger(CompletionTrigger),
+    /// A completion was manually requested (c-x)
+    Manual,
+    /// Some text was deleted and the cursor is now at `pos`
+    DeleteText { pos: usize },
+    /// Invalidate the current auto completion trigger
+    Cancel,
+}
```

Comment:
I need to look at the code again here, I think I just fogot to update the manual varaint while working on this.

Manual needs specical handling because we don't want to report a trigger char to the LSP here (that's how its intended by the LSP standard).

Reply:
I updated the code to make it clearer what happens here and better handle edgecases